pkg/controller/jobframework/interface.go

```diff
@@ -122,7 +122,7 @@ func QueueNameForObject(object client.Object) string {
 		return queueLabel
 	}
 	// fallback to the annotation (deprecated)
-	return object.GetAnnotations()[constants.QueueAnnotation]
```

Comment:
@alculquicondor 
First of all, I don't have any objections to dropping annotation support since we make annotation support as deprecated when changes in API versions occur (alpha -> beta)
However, IIUC, in k/k, those deprecated labels and annotations were never removed like this: https://github.com/kubernetes/kubernetes/blob/b722d017a34b300a2284b890448e5a605f21d01e/pkg/apis/batch/types.go#L41

For such a situation, is there any deprecation guideline?

Reply:
https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-parts-of-the-api

The policy actually states to only remove when incrementing versions. We could do that when we introduce v1beta2 or v1.

Reply:
Sounds good. @alexandear is there a way to add an exception in the linter for this deprecation?

Reply:
@alculquicondor yeah, it's possible

Reply:
So, you want me to only fix the deprecation comment and not change the code that uses `constants.QueueAnnotation`. Is that correct?

Reply:
Correct. We are using it when we are referring to an annotation, as far as I can see.

Reply:
Done.